src/lib/Libifl/pbsD_connect.c

```diff
@@ -800,6 +800,12 @@ __pbs_connect_extend(char *server, char *extend_data)
 		}
 
 		reply = PBSD_rdrpy(out);
+		if (reply == NULL) {
+			/* failed to read reply from the given connection point,
+			 * connection might be dropped by the TCP stack.
+			 */
+			return -1;
```

Comment:
It would be good to close the connection before returning. Add:
              CLOSESOCKET(connection[out].ch_socket);
               connection[out].ch_inuse = 0;

before the return -1;

Also add the same for the other -1 returns that are there a few lines above as well.